### PR TITLE
feat: Update azure-bootstrap and azure-identity in preparation for CAPI release

### PR DESCRIPTION
**Pull request by maciaszczykm**
## Summary
Extract azure-bootstrap and azure-identity changes from https://github.com/pluralsh/plural-artifacts/pull/741 in preparation for CAPI release. Additionally bump azure-bootstrap `provider_vsn` and update `sku_tier` to be compatible with it.

## Test Plan
```sh
export EXP_PLURAL_CAPI=false
plural init
plural bundle install bootstrap azure-k8s
plural build
plural link terraform bootstrap --name azure-bootstrap --path ../plural-artifacts/bootstrap/terraform/azure-bootstrap
plural link helm bootstrap --name azure-identity --path ../plural-artifacts/bootstrap/helm/azure-identity
plural build

# Set following required_providers in bootstrap/terraform/main.tf. It's required as provider_vsn will not be used during local link.
#   required_providers {
#    azurerm = {
#      source = "hashicorp/azurerm"
#      version = "~> 3.70.0"
#    }
#    kubernetes = {
#      source  = "hashicorp/kubernetes"
#      version = "~> 2.23.0"
#    }
#  }

plural deploy --commit "deploy"
plural destroy
```

```sh
export EXP_PLURAL_CAPI=false
plural init
plural bundle install bootstrap azure-k8s
plural build
plural deploy

plural link terraform bootstrap --name azure-bootstrap --path ../plural-artifacts/bootstrap/terraform/azure-bootstrap
plural link helm bootstrap --name azure-identity --path ../plural-artifacts/bootstrap/helm/azure-identity
plural build

# Set following required_providers in bootstrap/terraform/main.tf. It's required as provider_vsn will not be used during local link.
#   required_providers {
#    azurerm = {
#      source = "hashicorp/azurerm"
#      version = "~> 3.70.0"
#    }
#    kubernetes = {
#      source  = "hashicorp/kubernetes"
#      version = "~> 2.23.0"
#    }
#  }

plural deploy  --commit "deploy"
plural destroy
```
## Checklist

- [ ]  No images hosted from dockerhub
- [ ]  Are dashboards present to understand the health of the application.  There **must** be at least 1 of these
    - [ ]  all databases should have dashboards
    - [ ]  ideally also have at least cpu/mem utilization dashboards for webserver tier of the app
    - [ ]  you can use `plural from-grafana` to convert a grafana dashboard found via google to our CRD
- [ ]  Are scaling runbooks present
    - [ ]  all databases **must** have scaling runbooks
    - [ ]  you can use the charts in `pluralsh/module-library` to accelerate this
- [ ]  do you need to add config overlays?
    - [ ]  inputing secrets
    - [ ]  configuring autoscaling
- [ ]  If there’s a web-facing component to the app, we need to support OIDC authentication and setting up private networks if no authentication option is viable
- [ ]  All major clouds must be supported
    - [ ]  Azure
    - [ ]  AWS
    - [ ]  GCP

### CODE REVIEW

File: bootstrap/terraform/azure-bootstrap/deps.yaml [hunk 1]
```diff
@@ -1,9 +1,10 @@
 apiVersion: plural.sh/v1alpha1
 kind: Dependencies
 metadata:
-  description: Creates a AKS cluster and prepares it for bootstrapping 
-  version: 0.1.28
+  description: Creates an AKS cluster and prepares it for bootstrapping
+  version: 0.3.4
 spec:
+  breaking: true
```
Comment (maciaszczykm) [inline bootstrap/terraform/azure-bootstrap/deps.yaml:7]: @michaeljguarino @DavidSpek should we keep this flag?

Comment (maciaszczykm) [reply bootstrap/terraform/azure-bootstrap/deps.yaml:7]: Also, should we raise major version after recent changes?

File: bootstrap/terraform/azure-bootstrap/deps.yaml [hunk 2]
```diff
@@ -16,6 +17,7 @@ spec:
     network: network
     kubelet_msi_id: kubelet_msi_id
     node_resource_group: node_resource_group
+    capz_assigned_identity_client_id: capz_assigned_identity_client_id
   provider_wirings:
-    cluster: module.azure-bootstrap.cluster
-  provider_vsn: "0.1.2"
+    cluster: module.azure-bootstrap-cluster-api.cluster
+  provider_vsn: "0.1.3"
```
Comment (maciaszczykm) [inline bootstrap/terraform/azure-bootstrap/deps.yaml:23]: This is preventing me from testing it locally with link.

File: bootstrap/terraform/azure-bootstrap/deps.yaml [hunk 3]
```diff
@@ -16,6 +17,7 @@ spec:
     network: network
     kubelet_msi_id: kubelet_msi_id
     node_resource_group: node_resource_group
+    capz_assigned_identity_client_id: capz_assigned_identity_client_id
   provider_wirings:
-    cluster: module.azure-bootstrap.cluster
```
Comment (DavidSpek) [inline bootstrap/terraform/azure-bootstrap/deps.yaml:20]: This change needs to be removed.

**Pull request merged**